rest/importtest/collections_import_test.go

```diff
@@ -410,6 +429,100 @@ func TestMultiCollectionImportRemoveCollection(t *testing.T) {
 	// there should be 10 documents datastore1_{10..19}
 	require.Equal(t, int64(docCount), rt.GetDatabase().DbStats.SharedBucketImport().ImportCount.Value())
 
+	// Get the import version from the persisted config to ensure it didn't change
+	updatedImportVersion := GetImportVersion(t, rt, dbName)
+	require.Equal(t, initialImportVersion, updatedImportVersion)
+
+}
+
+// TestImportVersionWriteVariations - ensure import version is updated for all different APIs that can modify the collection set
+func TestImportVersionWriteVariations(t *testing.T) {
+
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyDCP)
+	base.SkipImportTestsIfNotEnabled(t)
+	numCollections := 3
+	base.RequireNumTestDataStores(t, numCollections)
+
+	ctx := base.TestCtx(t)
+	testBucket := base.GetTestBucket(t)
+	defer testBucket.Close(ctx)
+
+	rtConfig := &rest.RestTesterConfig{
+		CustomTestBucket: testBucket.NoCloseClone(),
+		PersistentConfig: true,
+	}
+
+	rt := rest.NewRestTester(t, rtConfig)
+	defer rt.Close()
+
+	// Set up collection payloads for 1, 2, 3 collections
+	oneCollectionConfig := rest.GetCollectionsConfig(t, testBucket, 1)
+	oneCollectionConfigJSON, err := json.Marshal(oneCollectionConfig)
+	require.NoError(t, err)
+	twoCollectionConfig := rest.GetCollectionsConfig(t, testBucket, 2)
+	twoCollectionConfigJSON, err := json.Marshal(twoCollectionConfig)
+	require.NoError(t, err)
+	threeCollectionConfig := rest.GetCollectionsConfig(t, testBucket, 3)
+	threeCollectionConfigJSON, err := json.Marshal(threeCollectionConfig)
+	require.NoError(t, err)
+
+	// Insert with PUT and single collection
+	dbName := "db"
+	response := rt.SendAdminRequest(http.MethodPut, "/"+dbName+"/", fmt.Sprintf(collectionsDbConfig, testBucket.GetName(), string(oneCollectionConfigJSON)))
+	rest.RequireStatus(t, response, http.StatusCreated)
```

Comment:
you could construct this differently with helper functions: which might be more readable:

```suggestion
	dbconfig := rt.NewDbConfig()
	dbconfig.Scopes = rest.GetCollectionsConfig(t, testBucket, 1)
	rest.RequireStatus(rt.CreateDatabase("db"), dbconfig), http.StatusCreated)
```

Reply:
For all of these, I feel that using the helper functions makes it hard to verify that we're testing all the supported PUT/POST permutations for db config update, which is what I wanted to ensure coverage of in this test.